test_libs/pyspec/eth2spec/test/block_processing/test_process_early_derived_secret_reveal.py

```diff
@@ -0,0 +1,130 @@
+import pytest
+
+from eth2spec.test.helpers.custody import get_valid_early_derived_secret_reveal
+from eth2spec.test.context import with_phase1, spec_state_test, expect_assertion_error
+
+
+def run_early_derived_secret_reveal_processing(spec, state, randao_key_reveal, valid=True):
+    """
+    Run ``process_randao_key_reveal``, yielding:
+      - pre-state ('pre')
+      - randao_key_reveal ('randao_key_reveal')
+      - post-state ('post').
+    If ``valid == False``, run expecting ``AssertionError``
+    """
+    yield 'pre', state
+    yield 'randao_key_reveal', randao_key_reveal
+
+    if not valid:
+        expect_assertion_error(lambda: spec.process_early_derived_secret_reveal(state, randao_key_reveal))
+        yield 'post', None
+        return
+
+    spec.process_early_derived_secret_reveal(state, randao_key_reveal)
+
+    slashed_validator = state.validator_registry[randao_key_reveal.revealed_index]
+
+    if randao_key_reveal.epoch >= spec.get_current_epoch(state) + spec.CUSTODY_PERIOD_TO_RANDAO_PADDING:
+        assert slashed_validator.slashed
+        assert slashed_validator.exit_epoch < spec.FAR_FUTURE_EPOCH
+        assert slashed_validator.withdrawable_epoch < spec.FAR_FUTURE_EPOCH
+    # lost whistleblower reward
+    # FIXME: Currently broken because get_base_reward in genesis epoch is 0
```

Comment:
Why is base reward 0 at genesis epoch? total active balance should be non-zero as well as the effective balance of the validator